pkg/controller/endpointmonitor/endpointmonitor_controller.go

```diff
@@ -117,10 +117,18 @@ func (r *ReconcileEndpointMonitor) Reconcile(request reconcile.Request) (reconci
 	delay := time.Until(createTime.Add(config.GetControllerConfig().CreationDelay))
 
 	for index := 0; index < len(r.monitorServices); index++ {
-		monitor := findMonitorByName(r.monitorServices[index], monitorName)
+		monitor, err := findMonitorByName(r.monitorServices[index], monitorName)
+		// if there was some error while getting a monitor, re-queue it to try on the next reconcile iteration
+		if err != nil {
+			return reconcile.Result{RequeueAfter: RequeueTime}, err
+		}
 		if monitor != nil {
 			// Monitor already exists, update if required
 			err = r.handleUpdate(request, instance, *monitor, r.monitorServices[index])
+			if err != nil {
+				log.Errorf("Error while handling update: %s", err)
+				return reconcile.Result{RequeueAfter: RequeueTime}, err
+			}
```

Comment:
This is redundant since it already requeue's after create or update action:
https://github.com/stakater/IngressMonitorController/blob/master/pkg/controller/endpointmonitor/endpointmonitor_controller.go#L135